new/mode-agnostic-data-collection.md

```diff
@@ -0,0 +1,559 @@
+      "description": "Globally Unique ID for this message",
+      "type": "string",
+      "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$"
+    },
+    "message_version": {
+      "description": "Message Version",
+      "type": "string",
+      "enum": [
+        "0.1.1"
+      ]
+    },
+    "message_type": {
+      "description": "Message Type",
+      "type": "string"
+    },
+    "organization_name": {
+      "description": "It is the name of the org on which the run took place",
+      "type": ["string", "null"]
+    },
+    "parent_name": {
+      "description": "The name of the parent of the entity",
+      "type": "string"
+    },
+    "parent_type": {
+      "description": "The type of the parent of the entity",
+      "type": "string",
+       "enum": [
+        "bag",
+        "client",
+        "cookbook",
+        "container",
+        "environment",
+        "group",
+        "node",
+        "organization",
+        "role"]
+    },
+    "recorded_at": {
+      "description": "It is the ISO timestamp when the action happened",
+      "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-5][0-9]:[0-9]{2}Z$",
+      "type": "string"
+    },
+    "remote_hostname": {
+      "description": "The remote hostname which initiated the action",
+      "type": "string"
+    },
+    "remote_request_id": {
+      "description": "The request-id generate by the remote client",
+      "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$",
+      "type": "string"
+    },
+    "request_id": {
+      "description": "The Enterprise Chef Request ID",
+      "type": "string"
+    },
+    "requestor_name": {
+      "description": "The name of the client or user that initiated the action",
+      "type": "string"
+    },
+    "requestor_type": {
+      "description": "Was the requestor a client or user?",
+      "type": "string",
+      "enum": ["client", "user"]
+    },
+    "service_hostname": {
+      "description": "The FQDN of the Enterprise Chef server",
+      "type": "string"
+    },
+    "task": {
+      "description": "What action was performed?",
+      "type": "string",
+      "enum": ["associate", "create", "delete", "dissociate", "invite", "reject", "update"]
+    },
+    "user_agent": {
+      "description": "The User-Agent of the requestor",
+      "type": "string"
+    },
+    "data": {
+      "description": "The payload containing the entire request data",
+      "type": "object"
+    }
+  },
+  "required": [
+    "entity_name",
+    "entity_type",
+    "id",
+    "message_type",
+    "message_version",
+    "organization_name",
+    "recorded_at",
+    "remote_hostname",
+    "request_id",
+    "requestor_name",
+    "requestor_type",
+    "service_hostname",
+    "task",
+    "user_agent"
+  ],
+  "dependencies": {
+    "parent_name": ["parent_type"],
+    "parent_type": ["parent_name"]
+  },
+  "title": "ActionSchema",
+  "type": "object"
+}```
+
```

Comment:
GHFMD isn't treating this the way you intended.  May need a line break before the ```